util/systemd/systemd_linux.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package systemd
+
+import (
+	"log"
+	"sync"
+
+	"github.com/mdlayher/sdnotify"
+)
+
+// XXX(Xe): this may not be the best idea, but this should only be used spairingly as
+// is and Go doesn't have the concept of OS-specific struct members.
+var (
+	systemdLock  sync.Mutex
+	systemdNotif *sdnotify.Notifier = func() *sdnotify.Notifier {
+		result, err := sdnotify.New()
+		if err != nil {
+			log.Printf("XXX(Xe): sdnotify error: %v", err)
+			return nil
+		}
+		log.Printf("XXX(Xe): systemd-notify support enabled")
+		return result
+	}()
+)
+
+func Ready() {
+	if systemdNotif == nil {
```

Comment:
The *sdnotify.Notifier type's methods become no-ops when it's nil, so this isn't strictly necessary.